File: samples/java-play2/app/controllers/BaseApiController.java
package controllers;


import play.mvc.*;
import play.data.*;
import play.*;

import com.wordnik.swagger.core.util.JsonUtil;
import com.fasterxml.jackson.databind.ObjectMapper;

import java.util.*;
import java.io.StringWriter;

import models.*;

public class BaseApiController extends Controller {
	static JavaRestResourceUtil ru = new JavaRestResourceUtil();

	protected static ObjectMapper mapper = JsonUtil.getJsonMapper();
	
	public static Result JsonResponse(Object obj) {
		return JsonResponse(obj, 200);
	}

	public static Result JsonResponse(Object obj, int code) {
		StringWriter w = new StringWriter();
		try {
			mapper.writeValue(w, obj);
		} catch (Exception e) {
			// TODO: handle proper return code
			e.printStackTrace();
		}

		response().setContentType("application/json");
		response().setHeader("Access-Control-Allow-Origin", "*");
    response().setHeader("Access-Control-Allow-Methods", "GET, POST, DELETE, PUT");
    response().setHeader("Access-Control-Allow-Headers", "Content-Type, api_key, Authorization");

		return ok(w.toString());
	}
}

File: samples/scala-play2-no-format/app/controllers/BaseApiController.scala
package controllers

import play.api._
import play.api.mvc._
import play.api.data._
import play.api.data.Forms._
import play.api.data.format.Formats._
import play.api.Play.current

import value._
import api._
import com.wordnik.swagger.core.util.{ JsonUtil, RestResourceUtil }

import java.io.StringWriter

object BaseApiController {
  val mapper = JsonUtil.getJsonMapper
}

class BaseApiController extends Controller with RestResourceUtil {
  // APIs
  protected def JsonResponse(data: Object) = {
    val w = new StringWriter()

    BaseApiController.mapper.writeValue(w, data)

    val jsonValue: String = w.toString()
    new SimpleResult[String](header = ResponseHeader(200), body = play.api.libs.iteratee.Enumerator(jsonValue)).as("application/json")
      .withHeaders(("Access-Control-Allow-Origin", "*"))
  }

  protected def JsonResponse(data: Object, code: Int) = {
    val w = new StringWriter()

    BaseApiController.mapper.writeValue(w, data)

    val jsonValue: String = w.toString()
    new SimpleResult[String](header = ResponseHeader(code), body = play.api.libs.iteratee.Enumerator(jsonValue)).as("application/json")
      .withHeaders(("Access-Control-Allow-Origin", "*"))
  }
}
